packages/core-flows/src/definition/cart/workflows/create-carts.ts

```diff
@@ -26,9 +40,27 @@ export const createCartWorkflow = createWorkflow(
       findOrCreateCustomerStep({
         customerId: input.customer_id,
         email: input.email,
-      })
+      }),
+      validateVariantsExistStep({ variantIds })
     )
 
+    // TODO: This is on par with the context used in v1.*, but we can be more flexible.
+    const pricingContext = transform(
```

Comment:
What does the transform function do?

Reply:
The `transform` function was added as a middleware between steps where the input and output don't fit perfectly together. We don't want data transformations to live as separate steps because they are typically dependent on the specific flow of events in a workflow, so we added the transformer as a utility function that pipes the data to other steps in the correct shape.

Reply:
Imagine steps created independently of each other, e.g. as part of two different workflows, that you now want to use in a new third workflow. Ideally, you will just dump these steps in, but it is unlikely that you can do that, as they were created entirely independently of each other and typically as generic as possible.

Reply:
@olivermrbl ah ok got it, I just checked the code and I guess the main point here is that they are lazily executed, compared to eg. here just saying
`pricingContext = {currency_code: input.currency_code, region_id: region.id...}`